FirebaseStorage/Tests/SwiftIntegration/StorageIntegration.swift

```diff
@@ -357,6 +409,33 @@ class StorageIntegration: XCTestCase {
     waitForExpectations()
   }
 
+  func testUnauthenticatedSimpleGetFileWithCompletion() throws {
+    let expectation = self.expectation(description: #function)
+    let ref = storage.reference(withPath: "ios/public/cookie")
+    let tmpDirURL = URL(fileURLWithPath: NSTemporaryDirectory())
+    let fileURL = tmpDirURL.appendingPathComponent("cookie.txt")
+    let data = try XCTUnwrap("Here's a 🍪, yay!".data(using: .utf8), "Data construction failed")
+
+    ref.putData(data, metadata: nil, completion: { metadata, error in
+      XCTAssertNotNil(metadata, "Metadata should not be nil")
+      XCTAssertNil(error, "Error should be nil")
+
+      ref.write(toFile: fileURL) { url, error in
+        XCTAssertNil(error, "Error should be nil")
+        do {
+          let url = try XCTUnwrap(url, "Failed to unwrap url")
+          XCTAssertEqual(fileURL, url)
+          let stringData = try String(contentsOf: fileURL, encoding: .utf8)
+          XCTAssertEqual(stringData, "Here's a 🍪, yay!")
+          expectation.fulfill()
+        } catch {
+          XCTFail("Throw in url completion block")
```

Comment:
Instead of wrapping this in a `do/catch` block you should be able to just throw since the function already mentions it can throw (on line 412).

Reply:
@ryanwilson  I'm a little confused so I think I'm missing something. How do I throw the error if it happens in the closure? I don't think the closure is defined to throw right? I was following along what was done in this method [here](https://github.com/firebase/firebase-ios-sdk/blob/b90ddf1339c8ec4b2004faf416f100beb59d0ccc/FirebaseStorageSwift/Tests/Integration/StorageIntegration.swift#L375)

Reply:
My bad, I was the confused one :) you're right, thanks. Maybe updating the comment to something like "Could not get String contents for data fetched" or something?